examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+  // Helmholtz equation with an iterative solver. The shifted Laplacian
+  // multigrid method is a well known approach to precondition this system, but
+  // this is beyond the scope of this tutorial.
+  template <int dim>
+  void ElasticWave<dim>::solve()
+  {
+    TimerOutput::Scope              t(computing_timer, "solve");
+    LinearAlgebraPETSc::MPI::Vector completely_distributed_solution(
+      locally_owned_dofs, mpi_communicator);
+
+    SolverControl                    solver_control;
+    PETScWrappers::SparseDirectMUMPS solver(solver_control, mpi_communicator);
+    solver.solve(system_matrix, completely_distributed_solution, system_rhs);
+
+    pcout << "   Solved in " << solver_control.last_step() << " iterations."
+          << std::endl;
+    constraints.distribute(completely_distributed_solution);
+    locally_relevant_solution = completely_distributed_solution;
+  }
+
+  // @sect4{ElasticWave::set_position_vector}
+
+  // We use this function to calculate the values of the position vector.
+  template <int dim>
+  void ElasticWave<dim>::set_position_vector()
+  {
+    Point<dim> p;
+    for (unsigned int position_idx = 0;
+         position_idx < parameters.nb_probe_points;
+         ++position_idx)
+      {
+        // Because of the way the operator + and - are overloaded. To substract
+        // two points, the following has to be done:
+        // `Point_b<dim> + (-Point_a<dim>)`
+        p = (position_idx / ((double)(parameters.nb_probe_points - 1))) *
+              (parameters.probe_stop_point + (-parameters.probe_start_point)) +
+            parameters.probe_start_point;
+        position[position_idx][0] = p[0];
+        position[position_idx][1] = p[1];
+        if (dim == 3)
+          {
+            position[position_idx][2] = p[2];
+          }
+      }
+  }
+
+  // @sect4{ElasticWave::store_frequency_step_data}
+
+  // This function stores in the HDF5 file the measured energy by the probe.
+  template <int dim>
+  void
+  ElasticWave<dim>::store_frequency_step_data(const unsigned int &frequency_idx)
+  {
+    TimerOutput::Scope t(computing_timer, "store_frequency_step_data");
+
+    // We store the displacement in the $x$ direction; the displacement in the
+    // $y$ direction is negligible.
+    const int probe_displacement_component = 0;
+
+    // The vector coordinates contains the coordinates in the HDF5 file of the
+    // points of the probe that are located in locally owned cells. The vector
+    // displacement_data contains the value of the displacement at these points.
+    std::vector<hsize_t>              coordinates;
+    std::vector<std::complex<double>> displacement_data;
+    for (unsigned int position_idx = 0;
+         position_idx < parameters.nb_probe_points;
+         ++position_idx)
+      {
+        Point<dim> point;
+        for (unsigned int dim_idx = 0; dim_idx < dim; ++dim_idx)
+          {
+            point(dim_idx) = position[position_idx][dim_idx];
```

Comment:
```suggestion
            point[dim_idx] = position[position_idx][dim_idx];
```
(For consistency with the other places where you index into a `Point`.)